public/skin/frontend/base/default/js/opcheckout.js

```diff
@@ -23,211 +24,256 @@ Checkout.prototype = {
+        let container;
         if (step) {
             if (this.loadWaiting) {
                 this.setLoadWaiting(false);
             }
-            container = $(step+'-buttons-container');
-            container.addClassName('disabled');
-            container.setStyle({opacity:.5});
+            container = document.getElementById(`${step}-buttons-container`);
+            container.classList.add('disabled');
+            container.style.opacity = 0.5;
             this._disableEnableAll(container, true);
-            Element.show(step+'-please-wait');
+            document.getElementById(`${step}-please-wait`).style.display = 'block';
         } else {
             if (this.loadWaiting) {
-                container = $(this.loadWaiting+'-buttons-container');
-                var isDisabled = (keepDisabled ? true : false);
+                container = document.getElementById(`${this.loadWaiting}-buttons-container`);
+                const isDisabled = keepDisabled ? true : false;
                 if (!isDisabled) {
-                    container.removeClassName('disabled');
-                    container.setStyle({opacity:1});
+                    container.classList.remove('disabled');
+                    container.style.opacity = 1;
                 }
                 this._disableEnableAll(container, isDisabled);
-                Element.hide(this.loadWaiting+'-please-wait');
+                document.getElementById(`${this.loadWaiting}-please-wait`).style.display = 'none';
             }
         }
         this.loadWaiting = step;
-    },
-
-    gotoSection: function (section, reloadProgressBlock) {
+    }
 
+    gotoSection(section, reloadProgressBlock) {
         if (reloadProgressBlock) {
             this.reloadProgressBlock(this.currentStep);
         }
         this.currentStep = section;
-        var sectionElement = $('opc-' + section);
-        sectionElement.addClassName('allow');
-        this.accordion.openSection('opc-' + section);
-        if(!reloadProgressBlock) {
+        const sectionElement = document.getElementById(`opc-${section}`);
+        sectionElement.classList.add('allow');
+        this.accordion.openSection(`opc-${section}`);
+        if (!reloadProgressBlock) {
             this.resetPreviousSteps();
         }
-    },
-
-    resetPreviousSteps: function () {
-        var stepIndex = this.steps.indexOf(this.currentStep);
-
-        //Clear other steps if already populated through javascript
-        for (var i = stepIndex; i < this.steps.length; i++) {
-            var nextStep = this.steps[i];
-            var progressDiv = nextStep + '-progress-opcheckout';
-            if ($(progressDiv)) {
-                //Remove the link
-                $(progressDiv).select('.changelink').invoke('remove');
-                $(progressDiv).select('dt').invoke('removeClassName','complete');
-                //Remove the content
-                $(progressDiv).select('dd.complete').invoke('remove');
+    }
```

Comment:
What do you think about merging the overridden method from here? https://github.com/MahoCommerce/maho/blob/main/public/skin/frontend/rwd/default/js/opcheckout_rwd.js

I see no harm in adding the `opc-has-progressed-from-login` class outside of rwd. But would just need to wrap the match media part since `bp` is defined in rwd's `app.js`.

```js
if (bp?.xsmall) {
    // Scroll viewport to top of checkout steps for smaller viewports
    if (window.matchMedia('(max-width: ' + bp.xsmall + 'px)').matches) {
        const checkoutSteps = document.getElementById('checkoutSteps');
        const topPosition = checkoutSteps.getBoundingClientRect().top + window.scrollY;
        window.scrollTo({ top: topPosition, behavior: 'smooth' });
    }
}
```

Reply:
And I think we can just use:
```js
checkoutSteps.scrollIntoView()
```

Reply:
@fballiano I totally agree with merging everything into base. Ideally `rwd` is maybe just a stylesheet and possibly some extra JS.

Reply:
exactly, I still want to review everything to see if there are some useless classes or something like that to clean, but ideally RWD is base...

Reply:
@justinbeaty I've done it in https://github.com/MahoCommerce/maho/pull/71#discussion_r1887071236 but actually I've removed the breakpoint check, it seems to work fine anyway also for desktop (and it kinda makes sense to scrollIntoView also on desktop)

Reply:
Nice!

Btw, any idea why those two script tags are in the template file instead of XML? I noticed it in #56 where I had to put the translation messages using `<layout handle="checkout_onepage_index">` instead of referencing the script.

Reply:
yeah I really don't know, I tried not to think about it when I saw that hahaha

let's see if we can fix that

Reply:
@justinbeaty done it in https://github.com/MahoCommerce/maho/pull/71/commits/cdd828247f1092cbe8480225b2a50d22b5469bdc but since opcheckout is not skin-related anymore, shouldn't we remove it from skin?

Reply:
@fballiano Yeah, I think it's better in `public/js`. 

Also, you can now change [Mage/Checkout/etc/jstranslator.xml](https://github.com/mahocommerce/maho/blob/main/app/code/core/Mage/Checkout/etc/jstranslator.xml) to:

```xml
<jstranslator>
    <script path="mage/opcheckout.js" module="checkout">
        <!-- ... -->
    </script>
</jstranslator>
```

Of course replace `mage/opcheckout.js` with wherever you put the file (note: there's no leading `js/`)

That way the strings are included whenever `opcheckout.js` is included.

Reply:
Looks good to me! And yeah, we should definitely standardize the JS folder's layout at some point...